Framework/PythonInterface/plugins/algorithms/SaveNexusPD.py

```diff
@@ -0,0 +1,312 @@
+#pylint: disable=no-init,invalid-name
+import mantid
+import mantid.simpleapi as api
+from mantid.kernel import StringListValidator
+import numpy as np
+
+try:
+    import h5py  # http://www.h5py.org/
+except ImportError:
+    pass
+
+class SaveNexusPD(mantid.api.PythonAlgorithm):
+    NX_CLASS = 'NX_class'
+    AXES_DICT = {'tof': 'time-of-flight',
+                 'dspacing': 'd-spacing',
+                 'Q': 'momentum transfer'}
+
+    _dtype = None
+    _compressArgs = {}
+    _tempNames = []
+    _sample = None
+    _sourcePos = None
+
+    def category(self):
+        return "DataHandling\\Nexus"
+
+    def name(self):
+        return "SaveNexusPD"
+
+    def summary(self):
+        return "Save powder diffraction data to a NeXus file"
+
+    def _determineDtype(self):
+        datatype = self.getProperty('DataType').value
+
+        if datatype == 'float32':
+            self._dtype = np.float32
+        elif datatype == 'float64':
+            self._dtype = np.float64
+        else:
+            raise RuntimeError("Do not understand '%s'" % datatype)
+
+    def _determineCompression(self):
+        compression = self.getProperty('Compression').value
+
+        if not str(compression) == "None":
+            self._compressArgs['compression'] = compression
+
+    def PyInit(self):
+        self.declareProperty(mantid.api.WorkspaceProperty('InputWorkspace',
+                                                          '',
+                                                          mantid.kernel.Direction.Input),
+                             "Workspace to save")
+        self.declareProperty(mantid.api.FileProperty('OutputFilename',
+                                                     '',
+                                                     action=mantid.api.FileAction.Save,
+                                                     extensions=['.h5']),
+                             doc='Name of the savefile')
+
+        group = 'Options'
+
+        self.declareProperty('NXentry', '',
+                             'Overrides the NXentry name from the workspace name')
+        self.declareProperty('DataType', 'float32',
+                             StringListValidator(['float32', 'float64']),
+                             doc='All data saved will be converted to this type')
+        self.declareProperty('Compression', 'gzip',
+                             StringListValidator(['gzip', 'lzw', 'None']),
```

Comment:
I think this should be `StringListValidator(['gzip', 'lzf', 'None'])`. The `h5py` lists the filter as `lzf` not `lzw`.